wavesexchange_topic/src/lib.rs

```diff
@@ -608,6 +702,71 @@ mod parse_and_format {
 
             Ok(())
         }
+
+        #[test]
+        fn pairs_one_test() -> anyhow::Result<()> {
+            let topic_data = Topic::parse_str("topic://pairs/amount_asset/price_asset")?.data();
+            let pairs = topic_data
+                .as_pairs()
+                .ok_or(anyhow::anyhow!("bad test case"))?;
+
+            assert_eq!(pairs[0].amount_asset, "amount_asset");
+            assert_eq!(pairs[0].price_asset, "price_asset");
+
+            Ok(())
+        }
+
+        #[test]
+        fn pairs_one_uri_only_test() -> anyhow::Result<()> {
+            let topic_data =
+                Topic::parse_str("topic://pairs/amount_asset/price_asset?pairs[]=skip/skip")?
+                    .data();
+            let pairs = topic_data
+                .as_pairs()
+                .ok_or(anyhow::anyhow!("bad test case"))?;
+
+            assert_eq!(pairs[0].amount_asset, "amount_asset");
+            assert_eq!(pairs[0].price_asset, "price_asset");
+
+            assert_eq!(pairs.len(), 1);
+
+            Ok(())
+        }
+
+        #[test]
+        fn pairs_many_test() -> anyhow::Result<()> {
+            let topic_data = Topic::parse_str("topic://pairs/?pairs[]=amount_asset/price_asset&pairs[]=amount_asset1/price_asset1")?.data();
+            let pairs = topic_data
+                .as_pairs()
+                .ok_or(anyhow::anyhow!("bad test case"))?;
+
+            assert_eq!(pairs[0].amount_asset, "amount_asset");
+            assert_eq!(pairs[0].price_asset, "price_asset");
+
+            assert_eq!(pairs[1].amount_asset, "amount_asset1");
+            assert_eq!(pairs[1].price_asset, "price_asset1");
+
+            Ok(())
+        }
+
+        #[test]
+        fn pairs_one_error_test() -> anyhow::Result<()> {
+            let topic_data = Topic::parse_str("topic://pairs/amount_asset");
+
+            assert!(topic_data.is_err());
+
+            Ok(())
+        }
+
+        #[test]
+        fn pairs_many_error_test() -> anyhow::Result<()> {
+            let topic_data =
+                Topic::parse_str("?pairs[]=amount_asset/price_asset&pairs[]=amount_asset1");
```

Comment:
Strange test case. I'd say it is better to write a single test fn for all incorrect cases, smth like `#[test] fn invalid_pairs_test` and maintain an array of different illegal URLs in that test fn.